### ISSUE TITLE
Cache vision

### ISSUE THREAD

**Issue by FurkanS1821** [LeagueSandbox/GameServer#621]
Vision should be cached somewhere on each tick start for performance, instead of being calculated when needed. Currently, half of the CPU operations are done in NavGrid (ray casting, getting if is walkable etc.)

**Comment by moonshadow565** [comment]
It could be done on each move which would match clients vision (onentervision and onleavevision packets).

**Comment by FurkanS1821** [comment]
Yeah that's a better idea.

**Comment by MythicManiac** [comment]
Yep, although I wouldn't mind a cache busting + recalculating function in case that's needed for some reason

**Comment by HarmfulBreeze** [comment]
Those methods should only be needed for debugging, we won't want them in production

**Comment by danil179** [comment]
Seems like #820 also reference some solution for the vision system performance .
